### ISSUE TITLE
document use of document$.subscribe()

### ISSUE THREAD

**Issue by alexvoss** [squidfunk/mkdocs-material#6196]
### Description

The documentation shows how to include custom JavaScript but does not say anything about how to register a callback that runs code when the page has loaded and the DOM is ready. Traditionally, this would be done using `document.ready()` but I seem to remember that JS frameworks have good reasons to modify this.

[Elsewhere in the documentation](https://squidfunk.github.io/mkdocs-material/reference/data-tables#sortable-tables), I can see that a function `document$.subscribe()` is used. This seems to be doing a similar thing but I could not figure out where the `document$` object comes from, so not sure. 

### Related links

- https://squidfunk.github.io/mkdocs-material/customization/?h=#additional-javascript
- https://squidfunk.github.io/mkdocs-material/?q=document$

### Proposed change

Add  a note about the use of a suitable callback mechanism. If, as I suspect, there is an interaction with the instant loading mechanism, this should also be documented.

### Before submitting

- [X] I have read and followed the [documentation issue guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-docs-issue/).
- [X] I have attached the links to the affected sections of [the documentation](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-docs-issue/#related-links)

**Comment by squidfunk** [comment]
Thanks for suggesting. Yes, this should be mentioned in the customization guide, which we need to rework, as I've already mentioned. I haven't found the time to do that yet, but I'll do it once I have some free time on my hands.

**Comment by alexvoss** [comment]
Can we add this one piece just now? If you point me in the right direction, I will do a PR.

**Comment by squidfunk** [comment]
Well, I'm not sure where to actually provide value with the current structure. Best location at the moment might be under Additional JavaScript in a tip box how to hook into navigation events and integrate with instant loading. Theme extension is about overriding the HTML templates (more involved) and theme development is about working with the code base of the theme itself (even more involved). Thus an tip box under Additional Javascript (collapsed) is likely the best idea.

**Comment by alexvoss** [comment]
Yep, can put it in there. Now I just need to understand where the `document$` object comes from and what friends it has. Just stumbled across `keyboard$` as well. You call that an observable - is it an [RxJS Observable](https://rxjs.dev/api/index/class/Observable)? Don't want to confidently march off in the wrong direction again...

**Comment by squidfunk** [comment]
It's exported by `bundle.js`:

https://github.com/squidfunk/mkdocs-material/blob/5c318799f694d10fc8507a5d6a449281e2bf526b/src/templates/assets/javascripts/bundle.ts#L306-L316

I wanted to add this to the documentation for a long time, since at least `document$` can be considered public API, but haven't managed to do so due to lack of time. I'd say we first focus on `document$`, and add better documentation for this in the future once the biggest change requests that are currently open are addressed, so probably later next year.

**Comment by squidfunk** [comment]
Oh, and yes, all of those are RxJS observables, except for `alert$`, which is a `Subject`. In fact, working with those observables allows to hook into several features of the theme, but as said, currently no bandwidth to write an article to explain it 😅 `alert$` is used as an example in our button documentation:

https://github.com/squidfunk/mkdocs-material/blob/5c318799f694d10fc8507a5d6a449281e2bf526b/docs/reference/buttons.md?plain=1#L48

**Comment by squidfunk** [comment]
Resolved via #6200 by @alexvoss
